posthog/tasks/process_scheduled_changes.py

```diff
@@ -0,0 +1,39 @@
+from posthog.models import ScheduledChange
+from django.utils import timezone
+from posthog.models import FeatureFlag
+from django.db import transaction, OperationalError
+
+models = {"FeatureFlag": FeatureFlag}
+
+
+def process_scheduled_changes() -> None:
+    try:
+        with transaction.atomic():
+            scheduled_changes = (
+                ScheduledChange.objects.select_for_update(nowait=True)
+                .filter(
+                    executed_at__isnull=True,
+                    scheduled_at__lt=timezone.now(),
+                )
+                .order_by("scheduled_at")
+            )
+
+            for scheduled_change in scheduled_changes.iterator(chunk_size=10000):
```

Comment:
this doesn't really work for us, because we can't use server-side cursors with pgbouncer's transaction pooling mode.

I'd suggest instead having a max limit of say 10,000 that every call processes. This can mean that if there are more than 10k scheduled changes, they'll get delayed until the next call 2mins later, which I think is fine.

Reply:
Also, did our model have an index on scheduled at / executed at?